### PR TITLE
Advanced routing

### PR DESCRIPTION
**Pull request by danhanson**
The routing server is specified within the application.conf with key routing.server which currently is sending to localhost:2000. The cluster router send a json post request to the server with datat that looks like this 

```
{ "commodityParameters":[],
  "vehicleParameters":[],
  "capacities":[],
  "commodities":{1:3, 2:4},
  "vehicles":[5],
  "distances":[[0,4766,6349,469,0],
               [4713,0,5805,4732,0],
               [6341,5850,0,6359,0],
               [445,4760,6343,0,0],
               [6607,6116,0,0,0]],
  "durations":[[0,371,459,90,0],
               [355,0,365,366,0],
               [469,390,0,481,0],
               [81,376,464,0,0],
               [512,433,0,0,0]],
   "objective":"0"
}
```

It then waits for a request that looks like this

```
{ "routes":[
      [5, 1, 2, 4, 3]
  ]
}
```

Where each route starts with the vehicle number and is a list of commodity starts and ends

### CODE REVIEW

**Comment by danhanson** [comment]
I didn't test response parsing yet

**Comment by danhanson** [comment]
I forgot to add capacities

File: conf/application.conf [hunk 1]
```diff
@@ -37,3 +34,5 @@ play.filters.cors {
 
 # The Global object provides hooks for application startup and shutdown.
 application.global=io.pathfinder.config.Global
+
+routing.server="http://0.0.0.0:2000"
```
Comment (aj-michael) [inline conf/application.conf]: Please fix this.

**Comment by aj-michael** [comment]
Please make capacities, commodityParameters and vehicleParameters arrays of objects.

**Comment by aj-michael** [comment]
Also, please make a Milestone tag to put this PR in. Steve checks those on Waffle and I like them.

**Comment by aj-michael** [comment]
The 1-routing looks good. Have you tested if this works manually?

**Comment by danhanson** [comment]
I tested the requests and they look right. I have not tested parsing the responses though

**Comment by danhanson** [comment]
oops I did not mean to add the default file thing

**Comment by aj-michael** [comment]
Tested this out with @danhanson, it works with v0.0.5 of csse497/PathfinderRouting

**Pull request merged**